# This file is part of HiGFXback

# requires
REQUIRES="make-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
sh ./Configure --prefix=

# build
make

# install
install -d $DESTDIR/bin
install samu $DESTDIR/bin

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/samurai-build.pc << EOF
Name: samurai
Version: 1.2
Description: Ninja-compatible build automation tool written in C
Requires: $REQUIRES

devel=\\
/bin/samu
EOF

exit

# patch
--- samurai-1.2.orig/Configure
+++ samurai-1.2/Configure
@@ -0,0 +1,13 @@
+#!/bin/sh
+
+PREFIX=/usr/local
+
+while [ $# != 0 ]; do
+  case $1 in
+    --prefix=*) PREFIX=${1#*=};;
+    *) echo "Unknown option: '$1'"; exit 1;;
+  esac
+  shift
+done
+
+sed -i "/^PREFIX/s|=.*|=$PREFIX|" Makefile
--- samurai-1.2.orig/Makefile
+++ samurai-1.2/Makefile
@@ -3,7 +3,7 @@
 PREFIX=/usr/local
 BINDIR=$(PREFIX)/bin
 MANDIR=$(PREFIX)/share/man
-ALL_CFLAGS=$(CFLAGS) -std=c99 -Wall -Wextra -Wpedantic -Wno-unused-parameter
+ALL_CFLAGS=$(CFLAGS) -std=c99 -Wall -Wextra -pedantic -Wno-unused-parameter
 OBJ=\
 	build.o\
 	deps.o\

# source
https://github.com/michaelforney/samurai/archive/1.2/samurai-1.2.tar.gz
